### DISCUSSION TITLE
Get the SharePoint "People or Group" field values and add them inside a folder's security/permission

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [pnp/pnpcore#1104] [Q&A] [answered]
I am working on building a Azure function which contain PnP Core SDK. the azure function need to do the following:-

loop through all list items with status = New >> for each list item >> create a new folder >> get the list item "People and Group" field values and add the values of the field inside the folder's security/permission.

Here is the custom list which contain a filed named "Folder Permission":-

![image](https://user-images.githubusercontent.com/79726511/219903428-af59c307-e28f-4b12-bd48-e11e82104478.png)

so what are the PnP Core SDK code to get the list item's People or Group values >> and add them to the created folder permission? the added entity can be User, SharePoint group, office 365 group, security group ?? any thing that can be added inside the "People or Group" field.. can anyone advice on this please?

Thanks

**Comment by jansenbe** [comment, marked as answer]
@mvcsharepointdev : you can get the user and groups added into that field as described here: https://pnp.github.io/pnpcore/using-the-sdk/listitems-fields.html#multi-user-fields. To add a folder check here: https://pnp.github.io/pnpcore/using-the-sdk/folders-intro.html#adding-folders. To then break permission inheritance of that folder and grant users permissions check https://pnp.github.io/pnpcore/using-the-sdk/security-intro.html#breaking-permission-inheritance and https://pnp.github.io/pnpcore/using-the-sdk/security-intro.html#assigning-roles-to-a-user-or-group
